docker/staging-scheduler/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM python:2.7-alpine
+
+WORKDIR /app
+RUN apk add curl
+
+COPY requirements.txt requirements.txt
+RUN pip install -r requirements.txt
+
+RUN mkdir -p /data
+RUN curl -Ls https://github.com/foursquare/luigi/archive/5ce6a271df9b42c7da9dd4ffc94b805ba7932d2b.zip > /data/luigi.zip
+RUN unzip /data/luigi.zip -d /data
+RUN rm /data/luigi.zip
+RUN mv /data/luigi-5ce6a271df9b42c7da9dd4ffc94b805ba7932d2b/ /app/luigi/
+COPY . .
```

Comment:
This is a smell. Conventionally, one would put the dockerfile at the top level of the repo, and copy in code or built artifacts from there.

Additionally, I see other room for improvement:
* We can reorder the dockerfile to put things that don't change often (e.g. the port exposure and entrypoint) early, achieving better layer caching
* We can pip install from the setup.py file, rather than using a requirements.

```dockerfile
FROM python:2.7-alpine

EXPOSE 8082
ENTRYPOINT [ "luigid" ]

RUN mkdir -p /luigi/state
VOLUME /luigi/state

RUN mkdir /etc/luigi
# IMO, we should inject this at docker or kubernetes runtime
ADD ./docker/staging-scheduler/etc/luigi/ /etc/luigi/
VOLUME /etc/luigi

WORKDIR /app/luigi

COPY . .
# We can install from the setup.py file, so we don't need to maintain a separate requirements file
RUN pip install -e .

```

Reply:
Got it running with `docker run -p 8082:8082 -v /Users/dshur/foursquare/docker-testing/client.cfg:/etc/luigi/client.cfg foursquare-luigi-setup` thanks for the tips, I have learned so much from this!

Reply:
Awesome